node/resolvers/Mutations/Settings.ts

```diff
@@ -0,0 +1,86 @@
+import {
+  B2B_SETTINGS_DATA_ENTITY,
+  B2B_SETTINGS_SCHEMA_VERSION,
+  B2B_SETTINGS_FIELDS
+} from '../../mdSchema'
+import GraphQLError from '../../utils/GraphQLError'
+import checkConfig from '../config'
+
+const B2BSettings = {
+  saveB2BSettings: async (
+    _: void,
+    {
+      input: { autoApprove, defaultPaymentTerms, defaultPriceTables }, 
+      page,
+      pageSize
+    }: {
+      input: B2BSettingsInput, page: number
+      pageSize: number
+    },
+    ctx: Context
+  ) => {    const {
+      clients: { masterdata },
+      vtex: { logger },
+    } = ctx
+
+    // create schema if it doesn't exist
+    await checkConfig(ctx)
+
+    try {
+      const b2bSettings = {
+        autoApprove,
+        defaultPaymentTerms,
+        defaultPriceTables
+      }
+      
+      const documentB2BSettingResult = await masterdata.searchDocuments({
```

Comment:
I think this create or update logic could be simplified a bit. Since there should only ever be one document in the settings table, we can hardcode an ID which allows us to do an upsert operation.

It'd look something like this:
```typescript
// at top level of file
const B2B_SETTINGS_DOCUMENT_ID = 'b2bSettings'

// Inside this try block
const result = await masterdata.createOrUpdateEntireDocument({
  id: B2B_SETTINGS_DOCUMENT_ID,
  dataEntity: B2B_SETTINGS_DATA_ENTITY,
  fields: b2bSettings,
  schema: B2B_SETTINGS_SCHEMA_VERSION
})
```

You could then update the getB2bSettings query to pull out this document by ID. That way it can return a single result instead of a list, and it should have better performance as well.